### DISCUSSION TITLE
Cannot get Frigate to record events

### DISCUSSION THREAD

**Discussion by dwrice0** [blakeblackshear/frigate#5937] [Ask A Question] [answered]
Frigate 0.11.1

I'm new to Frigate and I'm not sure what I've done.  However, I cannot get it to record Events anymore.  After the initial setup it recorded them fine but in the process of further customizing my yml file I must have done something (and didn't retain a previous copy of the file).  I have confirmed that it successfully detects a person by looking at the debug page for the cameras with bounding box turned on.

Here's my config file:
```
birdseye:
  enabled: True
  mode: continuous

mqtt:
  host: XXXXXX
  user: XXXXXX
  password: XXXXXX

detectors:
  cpu1:
    type: cpu
    num_threads: 3

record:
  enabled: True
  retain:
    days: 2
    mode: motion
  events:
    retain:
      default: 4
      mode: active_objects

objects:
  track:
    - person
    - bicycle
    - car
    - motorcycle
    - animal
    - dog

cameras:
  officecam:
    ffmpeg:
      inputs:
        - path: rtsp://XXXXXX:XXXXXX@XXXXXX/axis-media/media.amp?videocodec=h264
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 640
      height: 480
      fps: 6
  attic:
    ffmpeg:
      inputs:
        - path: rtsp://XXXXXX:XXXXXX@XXXXXX/axis-media/media.amp?videocodec=h264
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 640
      height: 480
      fps: 5
  testcam:
    ffmpeg:
      inputs:
        - path: rtsp://XXXXXX:XXXXXX@XXXXXX/cam/realmonitor?channel=1&subtype=0
          roles:
            - record
            - rtmp
        - path: rtsp://XXXXXX:XXXXXX@XXXXXX/cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
    detect:
      width: 640
      height: 480
      fps: 5

**Comment by NickM-27** [comment]
what is happening that makes you think things are not being recorded?

**Reply by dwrice0** [reply]
The event list is empty for the camera.  I believe the recordings exist but not the related event for the event list.

**Reply by NickM-27** [reply]
I would suggest enabling snapshots. Besides that, make sure you did not disable recordings / snapshots in the webUI (the button will be blue if it is on)

**Reply by dwrice0** [reply]
After adding the config to enable snapshots, it kind of worked once but then didn't on subsequent tests.  By "kinda", I mean that it showed the event in the cameras event list with a snapshot correctly showing the thumbnail of what caused the event.  However, the clip associated with the event did not actually contain the footage of what triggered the event (a person).  On all subsequent tests, it didn't register an event.  Here's my revised config file:

```
mqtt:
  host: XXXXXX
  user: XXXXXX
  password: XXXXXX

detectors:
  cpu1:
    type: cpu
    num_threads: 3

record:
  enabled: True
  retain:
    days: 2
    mode: motion
  events:
    retain:
      default: 4
      mode: active_objects

snapshots:
  enabled: True
  bounding_box: True

objects:
  track:
    - person
    - bicycle
    - car
    - motorcycle
    - animal
    - dog

cameras:
  officecam:
    ffmpeg:
      inputs:
        - path: rtsp://XXXXXX:XXXXXX@XXXXXX/axis-media/media.amp?videocodec=h264
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 640
      height: 480
      fps: 6
  attic:
    ffmpeg:
      inputs:
        - path: rtsp://XXXXXX:XXXXXX@XXXXXX/axis-media/media.amp?videocodec=h264
          roles:
            - detect
            - rtmp
            - record
    detect:
      width: 640
      height: 480
      fps: 5
  testcam:
    ffmpeg:
      inputs:
        - path: rtsp://XXXXXX:XXXXXX@XXXXXX/cam/realmonitor?channel=1&subtype=0
          roles:
            - record
            - rtmp
        - path: rtsp://XXXXXX:XXXXXX@XXXXXX/cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
    detect:
      width: 640
      height: 480
      fps: 5

**Reply by NickM-27** [reply]
okay, you may be running into a score issue. It's not really clear without a specific example of why an event would not be saved

**Reply by dwrice0** [reply]
As in not reaching the minimum to trigger the event? If so, what is the default minimum (for a person if they are different per object type)?

**Reply by NickM-27** [reply]
https://docs.frigate.video/guides/false_positives the default is 0.7

**Reply by dwrice0** [reply]
I lowered the min_score to 0.2 and threshold to 0.3 and did another test and it still failed to generate the event.

I started to wonder if this could be CPU load related since I'm running HA and Frigate on an Intel Atom based Supermicro server that is about 10 years old.  So, I dropped from 3 cameras to 1 camera and it now appears to be working correctly every time.  Do you think the CPU load theory holds water?  I ordered a TPU yesterday and it arrives tomorrow.  I guess we'll see.

Did a quick "w" from the Debian shell:
```
$ w
 20:34:21 up  4:07,  1 user,  load average: 1.03, 2.04, 2.80

**Reply by NickM-27** [reply]
what is your inference time in the system page?

**Reply by dwrice0** [reply]
![image](https://user-images.githubusercontent.com/52576449/230748474-1da3a9db-1e01-455b-bbb9-3cc18f6963f0.png)

So 1.5 seconds assuming that's measuring in ms. :)  I also assume that's not good based off of your explanation [here](https://github.com/blakeblackshear/frigate/discussions/3361).

**Reply by NickM-27** [reply, marked as answer]
yes, with an inference speed of 1.5 seconds that would 100% explain the issue

**Reply by dwrice0** [reply]
The reason it worked in the beginning was that I only had one camera in the system AND after adding the additional cameras, two of them would have been triggered for events at the same time during my testing.  Oof.

Just put a link to this discussion in your Recommended Hardware documentation section regarding getting the Coral TPU.

Looks like my CPU has 4 cores.  So, maybe with the other 3 defined in my config I might be able to get away with 3 cameras, but even that will likely struggle.

In any event, thanks for holding a noob's hand.

**Reply by dwrice0** [reply]
Just as a follow up.  Got the Coral USB installed and functioning today.  Here's my new inference speed:

![image](https://user-images.githubusercontent.com/52576449/230786147-5b14b94f-413c-4ac3-a5ae-2cc7a8d849f3.png)

Tested events when 2 cameras were triggered simultaneously and it worked perfectly.

**Reply by NickM-27** [reply]
Interesting, glad it's working better, if you have a usb3 port the speeds should be even better
